### ISSUE TITLE
react-pose - IE11 - 'Proxy' is undefined

### ISSUE THREAD

**Issue by genox** [Popmotion/popmotion#300]
Hi,

I get this error in IE11 when importing react-pose:

```
'Proxy' is undefined
ReferenceError: 'Proxy' is undefined
   at ./node_modules/react-pose/lib/posed.js (http://localhost:3000/_next/-/page/index.js:40737:1)
   at __webpack_require__ (http://localhost:3000/_next/-/manifest.js:715:13)
   at fn (http://localhost:3000/_next/-/manifest.js:118:13)
   at ./node_modules/react-pose/lib/index.js (http://localhost:3000/_next/-/page/index.js:40686:1)
   at __webpack_require__ (http://localhost:3000/_next/-/manifest.js:715:13)
   at fn (http://localhost:3000/_next/-/manifest.js:118:13)
   at Anonymous function (http://localhost:3000/_next/-/page/index.js:64188:22)
   at ./src/sections/header/menu.tsx (http://localhost:3000/_next/-/page/index.js:64176:29)
   at __webpack_require__ (http://localhost:3000/_next/-/manifest.js:715:13)
   at fn (http://localhost:3000/_next/-/manifest.js:118:13)
```

As far as I could figure out, Proxy cannot be easily polyfilled. Neither TS nor babel provides an easy fix. Any ideas?

**Comment by InventingWithMonster** [comment]
Ah yeah sorry React Pose doesn’t support IE11

**Comment by genox** [comment]
That’s too bad. But I can understand. It’s a PITA.. I solved it by polyfilling the library with GoogleChrome’s polyfill and then conditionally disabling UI animations for IE11 specifically. This allows for graceful degradation. The polyfill is only necessary to avoid having dynamic imports. Posed is never called. Unfortunstely the polyfill itself doesn’t make posed work magically in IE11 but it prevents the app from bailing.

**Comment by InventingWithMonster** [comment]
Nice one thanks. When I get to an FAQs I’ll add this

**Comment by hilary-L** [comment]
@genox @InventingWithMonster Ran into this issue myself on a recent app. Chose not to use the google chrome polyfill, but rather to do code splitting at the route level. In this particular app I'm only using React Pose animations on routes that occur subsequent to the initial page load. I made two versions of the components that have React Pose animations, with and without the animations. I then used code splitting with webpack to create different chunks for these components, and did a `if(typeof Proxy !== 'undefined)` to determine which async component to render. The app now works fine in IE 11 (with no animations), and I've preserved support for the animations in all modern browsers.
